xarray/core/groupby.py

```diff
@@ -310,6 +306,27 @@ def __len__(self):
     def __iter__(self):
         return zip(self._unique_coord.values, self._iter_grouped())
 
+    def _get_index_and_items(self, index, grouper):
+        from .resample_cftime import CFTimeGrouper
+        s = pd.Series(np.arange(index.size), index)
+        if isinstance(grouper, CFTimeGrouper):
+            first_items = grouper.first_items(index)
+            full_index = first_items.index
+            if first_items.isnull().any():
```

Comment:
if you merge in master against, you could switch this block back to using `Series.dropna()`.

Reply:
Done.